Source/loadsave.cpp

```diff
@@ -2326,8 +2326,6 @@ void SaveGameData(MpqWriter &saveWriter)
 	file.WriteBE<uint32_t>(static_cast<uint32_t>(std::min(Missiles.size(), MaxMissilesForSaveGame)));
 	file.WriteBE<int32_t>(ActiveObjectCount);
 
-	leveltype = GetLevelType(currlevel);

```

Comment:
This was added for correcting crypt and hive being designated cathedral and caves in original hellfire save games. Do we not still need it for correcting this? Maybe we should just add `(!setLevel)` before it?

Reply:
When saving data we need to stay compatible with the old format.
This is done by calling `getHellfireLevelType` when writing to the save game.
The in-memory data is still in the new format (`DTYPE_CRYPT`&`DTYPE_NEST`).
And I thought saving shouldn't change the in-memory data or?

(When reading from the save game `(!setLevel)` is used).

Reply:
Oh wait, I must have looked in the wrong spot, I thought this was in the LoadLevel function, not save data...